lib/vector-core/src/event/vrl_target.rs

```diff
@@ -162,6 +172,30 @@ impl VrlTarget {
     }
 }
 
+fn set_metric_tag_values(name: String, value: &Value, metric: &mut Metric, multi_value_tags: bool) {
+    if multi_value_tags {
+        let tag_values = value
+            .as_array()
+            .unwrap_or(&[])
+            .iter()
+            .filter_map(|value| match value {
+                Value::Bytes(bytes) => {
+                    Some(TagValue::Value(String::from_utf8_lossy(bytes).to_string()))
+                }
+                Value::Null => Some(TagValue::Bare),
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+
+        metric.set_multi_value_tag(name, tag_values);
+    } else {
+        // set a single tag value
+        if let Ok(tag_value) = value.try_bytes_utf8_lossy().map(Cow::into_owned) {
```

Comment:
Should we try to handle bare tag assignments in "single" mode as well, or would that constitute a breaking change?

Reply:
That seems reasonable. I added it.